crates/rpc/rpc-types-compat/src/transaction.rs

```diff
@@ -46,3 +50,86 @@ pub trait TransactionCompat: Send + Sync + Unpin + Clone + Debug {
         request: TransactionRequest,
     ) -> Result<TxTy<Self::Primitives>, Self::Error>;
 }
+
+/// Converts `self` into `T`.
+pub trait IntoRpcTx<T> {
+    /// Performs the conversion.
+    fn into_rpc_tx(self, tx_info: TransactionInfo) -> T;
+}
+
+/// Converts [`TransactionRequest`] into `Self`.
+pub trait FromTxReq {
+    /// Performs the conversion.
+    fn from_tx_req(request: TransactionRequest) -> Result<Self, ValueError<TransactionRequest>>
+    where
+        Self: Sized;
+}
+
+impl IntoRpcTx<Transaction> for Recovered<EthereumTxEnvelope<TxEip4844>> {
+    fn into_rpc_tx(self, tx_info: TransactionInfo) -> Transaction {
+        Transaction::from_transaction(
+            self.map(|v| match v {
+                EthereumTxEnvelope::Eip4844(v) => {
+                    EthereumTxEnvelope::Eip4844(v.map(TxEip4844Variant::TxEip4844))
+                }
+                EthereumTxEnvelope::Legacy(v) => EthereumTxEnvelope::Legacy(v),
+                EthereumTxEnvelope::Eip2930(v) => EthereumTxEnvelope::Eip2930(v),
+                EthereumTxEnvelope::Eip1559(v) => EthereumTxEnvelope::Eip1559(v),
+                EthereumTxEnvelope::Eip7702(v) => EthereumTxEnvelope::Eip7702(v),
+            }),
+            tx_info,
+        )
+    }
+}
+
+impl FromTxReq for EthereumTxEnvelope<TxEip4844> {
+    fn from_tx_req(request: TransactionRequest) -> Result<Self, ValueError<TransactionRequest>> {
+        TransactionRequest::build_typed_simulate_transaction(request)
+    }
+}
+
+/// Error that occurred during conversions into RPC response.
+#[derive(Debug, Clone, Error)]
+pub enum CompatError {
+    /// Error that happens on conversion into transaction RPC response.
+    #[error("Failed to convert transaction into RPC response")]
+    TransactionConversionError,
+}
```

Comment:
I think we can replace this with a string error so that we can directly convert this into a errorObject

or we use 

https://github.com/paradigmxyz/reth/blob/8e54c4b2a626c1fb51ad9741e4c29117499444c4/crates/rpc/rpc-eth-types/src/error/mod.rs#L149-L151